src/broadcast/scrobblingmanager.cpp

```diff
@@ -0,0 +1,210 @@
+#include <QObject>
+
+#include "broadcast/scrobblingmanager.h"
+#include "broadcast/filelistener.h"
+#include "control/controlproxy.h"
+#include "engine/enginexfader.h"
+#include "mixer/deck.h"
+#include "mixer/playerinfo.h"
+#include "mixer/playermanager.h"
+
+ScrobblingManager::ScrobblingManager(PlayerManager *manager) :
+m_CPGuiTick("[Master]", "guiTick50ms",this),
+m_CPCrossfader("[Master]","crossfader", this),
+m_CPXFaderCurve(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                "xFaderCurve"),this),
+
+m_CPXFaderCalibration(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                      "xFaderCalibration"),this),
+
+m_CPXFaderMode(ConfigKey(EngineXfader::kXfaderConfigKey, 
+               "xFaderMode"),this),
+
+m_CPXFaderReverse(ConfigKey(EngineXfader::kXfaderConfigKey, 
+                  "xFaderReverse"),this),
+
+m_pManager(manager),
+m_broadcaster(new TrackTimers::ElapsedTimerQt)
+{
+    m_CPGuiTick.connectValueChanged(SLOT(slotGuiTick(double)));
+    connect(&PlayerInfo::instance(),SIGNAL(currentPlayingTrackChanged(TrackPointer)),
+            &m_broadcaster,SLOT(slotNowListening(TrackPointer)));
+    m_broadcaster
+        .addNewScrobblingService(new FileListener("nowListening.txt"));
```

Comment:
I think it is: addNewCurrentTrackConsumer() or so. I think it is better to call this form outside the class. The manager should have no knowledge which type of consumers are connected. It should only track a list of generic consumers if required.

Reply:
The file should be finally written into a user directory.

Reply:
Yeah, that's what I was thinking too, but since the user will choose the services this is a WIP solution. The file path is WIP too because the user should be able to choose the path of the file.